File: backend/app/phone.py
import re


def is_valid_phone(phone: str) -> bool:
    """בדיקה בסיסית - לא מאמתת שהמספר באמת קיים, רק שהצורה הגיונית
    (רצף ספרות סביר, עם או בלי + מוביל, 9-15 ספרות)"""
    digits = re.sub(r"[^\d+]", "", phone)
    return bool(re.fullmatch(r"\+?\d{9,15}", digits))

File: frontend/lib/phone.ts
/**
 * wa.me דורש מספר בפורמט בינלאומי, ספרות בלבד, בלי + או 0 מוביל.
 * הופך "050-1234567" / "0501234567" ל-"972501234567".
 */
export function formatPhoneForWhatsapp(phone: string): string {
  const digits = phone.replace(/\D/g, "");
  if (digits.startsWith("972")) return digits;
  if (digits.startsWith("0")) return "972" + digits.slice(1);
  return digits;
}
